modules/apps/transfer/types/trace.go

```diff
@@ -70,6 +75,36 @@ func (dt DenomTrace) GetFullDenomPath() string {
 	return dt.GetPrefix() + dt.BaseDenom
 }
 
+// extractPathAndBaseFromFullDenom returns the trace path and the base denom from
+// the elements that constitute the complete denom.
+func extractPathAndBaseFromFullDenom(fullDenomItems []string) (string, string) {
+	var (
+		path      []string
+		baseDenom []string
+	)
+
+	length := len(fullDenomItems)
+	for i := 0; i < length; i = i + 2 {
```

Comment:
do we validate anywhere that the length is even? would be better if we group `fullDenomItems[i]` and `[i+1]` into a single obj, but if this is directly copied from the open sourced version then we can leave it

Reply:
@codchen there is a check right below this to verify `i < length-1 && i > 2` https://github.com/sei-protocol/sei-ibc-go/blob/sei-ibc-v3.2.0/modules/apps/transfer/types/trace.go#L97 before accessing `fullDenomItems[i]` and `[i+1]` and adding to the ibc `path`. 
Otherwise it gets used as the `baseDenom` https://github.com/sei-protocol/sei-ibc-go/blob/sei-ibc-v3.2.0/modules/apps/transfer/types/trace.go#L100 . 

It is common for `fullDenomItems` to have odd length - there are unit tests as well in https://github.com/sei-protocol/sei-ibc-go/blob/sei-ibc-v3.2.0/modules/apps/transfer/types/trace_test.go#L9 where this occurs.
Let me know if that answers your point. And yea this is taken directly from open source, could be cleaner but will leave this as is for now.